app/src/main/java/com/beemdevelopment/aegis/otp/GoogleAuthInfo.java

```diff
@@ -343,5 +353,30 @@ public int getBatchIndex() {
         public int getBatchId() {
             return _batchId;
         }
+
+        public static List<Integer> getMissingIndexes(@NonNull List<Export> exports) throws IllegalArgumentException {
+            if (Objects.isNull(exports) || exports.isEmpty()) {
+                throw new IllegalArgumentException("Provided list was null or empty.");
+            }
+            int batchId = exports.get(0).getBatchId();
+            int batchSize = exports.get(0).getBatchSize();
+            List<Integer> indexesPresent = new ArrayList<>();
+            List<Integer> indexesMissing = new ArrayList<>();
+
+            for (Export export: exports) {
+                if (export.getBatchId() == batchId) {
+                    indexesPresent.add(export.getBatchIndex());
+                } else {
+                    throw new IllegalArgumentException("Export contains information for an unrelated batch. Try importing 1 batch at a time.");
```

Comment:
It's good to do this sanity check here, but I'd also like to have the caller check for this before calling this function, so that we can show a translatable error message. I'd suggest adding a separate static function to this class to check whether all QR codes are from a single batch. Something like:

```java
public static boolean isSingleBatch(@NonNull List<Export> exports) {
    if (exports.isEmpty()) {
        return true;
    }

    int batchId = exports.get(0).getBatchId();
    for (Export export : exports) {
        if (export.getBatchId() != batchId) {
            return false;
        }
    }

    return true;
}
```

When calling it from MainActivity, show a translatable error dialog if needed with the message you're using for the Exception now. The Exception message can be simplified to something like "Export list contains entries from different batches".